File: scala/sourcecode/src/main/scala/com/tersesystems/echopraxia/sapi/sourcecode/LoggerFactory.scala
package com.tersesystems.echopraxia.sapi.sourcecode

import com.tersesystems.echopraxia.core.{Caller, CoreLoggerFactory}
import com.tersesystems.echopraxia.sapi.FieldBuilder

import support._

/**
 * LoggerFactory for a logger with source code enabled.
 */
object LoggerFactory {
  val FQCN: String = classOf[DefaultLoggerMethods[_]].getName

  val fieldBuilder: FieldBuilder = new FieldBuilder {}

  def getLogger(name: String): Logger[FieldBuilder] = {
    val core = CoreLoggerFactory.getLogger(FQCN, name)
    new Logger(core, fieldBuilder)
  }

  def getLogger(clazz: Class[_]): Logger[FieldBuilder] = {
    val core = CoreLoggerFactory.getLogger(FQCN, clazz.getName)
    new Logger(core, fieldBuilder)
  }

  def getLogger: Logger[FieldBuilder] = {
    val core = CoreLoggerFactory.getLogger(FQCN, Caller.resolveClassName)
    new Logger(core, fieldBuilder)
  }

}

File: simple/src/main/java/echopraxia/simple/LoggerFactory.java
package echopraxia.simple;

import echopraxia.logging.spi.CoreLoggerFactory;
import org.jetbrains.annotations.NotNull;

public class LoggerFactory {

  private static final @NotNull String FQCN = "echopraxia.simple.Logger";

  public Logger getLogger(String name) {
    var core = CoreLoggerFactory.getLogger(FQCN, name);
    return new Logger(core);
  }

  public Logger getLogger(Class<?> clazz) {
    var core = CoreLoggerFactory.getLogger(FQCN, clazz);
    return new Logger(core);
  }
}
